src/Command/Ingester/CreateUserCommand.php

```diff
@@ -0,0 +1,461 @@
+use Symfony\Component\Console\Helper\ProgressBar;
+use OAT\SimpleRoster\Command\CommandProgressBarFormatterTrait;
+use OAT\SimpleRoster\Repository\Criteria\FindLineItemCriteria;
+use League\Csv\Writer;
+use Throwable;
+
+class CreateUserCommand extends Command
+{
+    use CommandProgressBarFormatterTrait;
+    public const NAME = 'roster:create-user';
+
+    private NativeUserRepository $userRepository;
+    private UserPasswordEncoderInterface $passwordEncoder;
+    private AssignmentIngester $assignmentIngester;
+    private LineItemRepository $lineItemRepository;
+    private LtiInstanceRepository $ltiInstanceRepository;
+    private Filesystem $filesystem;
+    protected ProgressBar $progressBar;
+
+    private const DEFAULT_BATCH_SIZE = '10';
+    private const OPTION_LINE_ITEM_IDS = 'line-item-ids';
+    private const OPTION_LINE_ITEM_SLUGS = 'line-item-slugs';
+    private const OPTION_GROUP_PREFIX = 'group-prefix';
+    private const DEFAULT_USERNAME_INCREMENT_NO = 0;
+
+    private SymfonyStyle $symfonyStyle;
+    /** @var string[] */
+    private array $lineItemSlugs;
+    private array $userPrefix;
+
+    /** @var int[] */
+    private array $lineItemIds;
+
+    /** @var int */
+    private int $batchSize;
+    private int $userGroupBatchCount = 0;
+    private int $groupIndex = 0;
+
+    public function __construct(
+        LineItemRepository $lineItemRepository,
+        AssignmentRepository $assignmentRepository,
+        LtiInstanceRepository $ltiInstanceRepository,
+        UserPasswordEncoderInterface $passwordEncoder,
+        NativeUserRepository $userRepository,
+        AssignmentIngester $assignmentIngester,
+        Filesystem $filesystem
+        ) {
+        parent::__construct(self::NAME);
+        $this->lineItemRepository = $lineItemRepository;
+        $this->assignmentRepository = $assignmentRepository;
+        $this->ltiInstanceRepository = $ltiInstanceRepository;
+        $this->passwordEncoder = $passwordEncoder;
+        $this->userRepository = $userRepository;
+        $this->assignmentIngester = $assignmentIngester;
+        $this->filesystem = $filesystem;
+    }
+
+    protected function configure(): void
+    {
+        parent::configure();
+
+        $this->setDescription('Automate a user generate list');
+
+        $this->addOption(
+            self::OPTION_LINE_ITEM_IDS,
+            'i',
+            InputOption::VALUE_REQUIRED,
+            'Comma separated list of line item IDs',
+        );
+
+        $this->addOption(
+            self::OPTION_LINE_ITEM_SLUGS,
+            's',
+            InputOption::VALUE_REQUIRED,
+            'Comma separated list of line item slugs',
+        );
+
+        $this->addOption(
+            'batch',
+            'b',
+            InputOption::VALUE_REQUIRED,
+            'User Create Batch size',
+            self::DEFAULT_BATCH_SIZE
+        );
+
+        $this->addArgument(
+            'user-prefix',
+            InputArgument::REQUIRED,
```

Comment:
You can probably use `InputArgument::IS_ARRAY | InputArgument::REQUIRED` - See [here](https://symfony.com/doc/current/console/input.html)

Then you can probably remove or simplify `initializeUserPrefixOption()`